### DISCUSSION TITLE
Are selectors suitable for high frequency updates like pointer events?

### DISCUSSION THREAD

**Discussion by elimist3** [facebookexperimental/Recoil#1694] [Q&A]
I have a drag and drop app I'm trying to implement with Recoil and everything has been going fine so far, except in one situation where I have an atom selector returning some derived state. 

The dependency chain looks like this: 

`Pointer move event --> atom<Data> --> selector<DerivedData> --> { multiple subscriber components }`

The atom itself gets set at quite a high rate - it basically tracks the mouse's coordinates plus some other metadata.  The downstream selector computes some state which multiple components rely on to determine whether or not the cursor location is currently within their boundaries. 

I'm wondering if this a bad use case for selectors because the current implementation feels very sluggish and suffers from dropped frames compared to the version that didn't use selector derived state. The react profiler seems to indicate that there are no excessive re-renders going on. 

Is this simply not what selectors were designed for?

**Comment by drarmstr** [comment]
Mouse movement is potentially a lot of reevaluations, but hard to say without profiling directly.  You may want to consider [adjusting the selectors to avoid caching.](https://recoiljs.org/docs/api-reference/core/selector#cache-policy-configuration)

**Reply by elimist3** [reply]
Thanks for the reply - I know it can be difficult to answer such a question since there can be many contributing factors, but just wanted to check if I'm using selectors in a way they were not meant to be used. 

I'll try adjusting the cache settings as you've suggested!
